docs/sources/writing-guide/documentation-structure/index.md

```diff
@@ -54,6 +54,26 @@ For more information about how to write tasks, refer to [Tasks]({{< relref "../t
 
 > **Note:** If a directory contains multiple pages or subdirectories, it is a branch bundle, and it must include an `_index.md` file. A directory containing only one page is a leaf bundle, and the content filename must be `index.md`.
 
-## Useful links
+## Pages and page bundles
 
-- For more information about branch bundles and leaf bundles, refer to [Page bundles](https://gohugo.io/content-management/page-bundles/).
+Each web page generated by Hugo comes from one of three source files:
+- `page/_index.md`: a Hugo branch bundle
+- `page/index.md`: a Hugo leaf bundle
+- `page.md`: a Hugo page
+
+Although each of the preceding examples results in the same URL (`/page/`), Hugo works with each source file differently.
+
+**Branch bundles** (`page/_index.md`) are required to produce page hierarchies.
+For the `/page/subpage/` URL to generated, there must first be a `page/_index.md` branch bundle source file.
+
+**Leaf bundles** (`page/index.md`) are required to conveniently and consistently bundle page assets.
+To refer to a style sheet `page/style.css` with the link `./style.css`, the link must be in a `page/index.md` leaf bundle source file.
+
+**Leaf bundles** (`page/index.md`) are also required to mount content from one part of the site to another using Hugo mounts.
+Hugo mounts use a virtual filesystem before site generation, and only directories can be mounted.
+To mount the page `/page/` to `/other/page/`, there must first be a `page/index.md` leaf bundle source file.
+
+**Pages** (`page.md`) are any source files that are not leaf or branch bundles.
+It is convenient to use pages when none the behaviors of leaf or branch bundles are required as it can be easier to distinguish two source files in some text editors or IDEs.
```

Comment:
```suggestion
It is convenient to use pages when none of the behaviors of leaf or branch bundles are required, as it can be easier to distinguish two source files in some text editors or IDEs.
```